suite-native/module-accounts-import/src/screens/ScanQRCodeModalScreen.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import { Camera, useCameraDevices } from 'react-native-vision-camera';
+import { Linking, SafeAreaView } from 'react-native';
+
+import { BarcodeFormat, useScanBarcodes } from 'vision-camera-code-scanner';
+
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { Box, IconButton, Text } from '@suite-native/atoms';
+import {
+    AccountsImportStackParamList,
+    AccountsImportStackRoutes,
+    StackProps,
+} from '@suite-native/navigation';
+
+const cameraWrapperStyle = prepareNativeStyle(_ => ({
+    position: 'relative',
+    overflow: 'hidden',
+    flex: 1,
+}));
+
+const cameraStyle = prepareNativeStyle(_ => ({
+    flex: 1,
+    height: '100%',
+}));
+
+const modalHeaderStyle = prepareNativeStyle(utils => ({
+    flexDirection: 'row',
+    width: '100%',
+    padding: utils.spacings.medium,
+    flex: 1,
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    position: 'absolute',
+    zIndex: 1,
+    top: 20,
+}));
+
+// TODO add navigation types
+export const ScanQRCodeModalScreen = ({
+    navigation,
+}: StackProps<AccountsImportStackParamList, AccountsImportStackRoutes.XpubScanModal>) => {
+    const [hasCameraPermission, setHasCameraPermission] = useState(false);
+    const { applyStyle } = useNativeStyles();
+    const devices = useCameraDevices();
+    const device = devices.back;
+
+    useEffect(() => {
+        const requestCamera = async () => {
```

Comment:
Can we put this function before component? Seems to have no dependency to any prop.

Reply:
It is calling `setHasCameraPermission` in the body of the function so it needs to be in same scope.